src/client/fontengine.cpp

```diff
@@ -59,7 +59,7 @@ FontEngine::FontEngine(gui::IGUIEnvironment* env) :
 		"mono_font_path", "mono_font_path_bold", "mono_font_path_italic",
 		"mono_font_path_bold_italic",
 		"fallback_font_path",
-		"dpi_change_notifier", "gui_scaling",
+		"dpi_change_notifier", "display_density_factor", "gui_scaling",
 	};
 
 	for (auto name : settings)
```

Comment:
I just noticed there's no unregister here :thinking:

Reply:
> I just noticed there's no unregister here 🤔

fixed that

> Maybe we should solve this using RAII

seems like a good idea to me, but too big for this PR (I assume you mean adding a helper class that "owns" a callback registration)

Reply:
I was thinking about this too and it's pretty straightforward to batch remove them (not RAII though) by specifying the userdata pointer, which is commonly `this` and removing assigned callbacks.